core/kernel/dt.c

```diff
@@ -51,6 +51,101 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
 	return prop;
 }
 
+int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop;
+	int len_prop;
+	int it_num;
+
+	/*
+	 * Interrupt property is defined with 3x32 bits word:
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 *  - Interrupt Trigger type
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if ((int_prop == NULL) || (len_prop < 2))
+		return -1;
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+
+	return it_num;
+}
+
+paddr_t dt_node_offset_by_compatible_status(void *fdt, int status, int *node,
+		const char *compatible)
+{
+	paddr_t offset = 0;
+	int fnode;
+	int node_status;
+
+	fnode = fdt_node_offset_by_compatible(fdt, *node,
+			compatible);
```

Comment:
Please align after the `(` at the line above, here and at other places in this file.